src/connectors/autocomplete/__tests__/connectAutocomplete-test.js

```diff
@@ -69,4 +70,84 @@ describe('connectAutocomplete', () => {
     expect(helper.search).toHaveBeenCalledTimes(1);
     expect(helper.getState().query).toBe('foo');
   });
+
+  it('with escapeHTML should escape the hits', () => {
+    const renderFn = jest.fn();
+    const makeWidget = connectAutocomplete(renderFn);
+    const widget = makeWidget({ escapeHTML: true });
+
+    const helper = jsHelper(fakeClient, '', {});
+    helper.search = jest.fn();
+
+    const hits = [
+      {
+        _highlightResult: {
+          foobar: {
+            value: `<script>${TAG_PLACEHOLDER.highlightPreTag}foobar${
+              TAG_PLACEHOLDER.highlightPostTag
+            }</script>`,
+          },
+        },
+      },
+    ];
+
+    const escapedHits = [
+      {
+        _highlightResult: {
+          foobar: {
+            value: '&lt;script&gt;<mark>foobar</mark>&lt;/script&gt;',
+          },
+        },
+      },
+    ];
+
+    widget.init({ helper, instantSearchInstance: {} });
+    const results = new SearchResults(helper.state, [{ hits }]);
+    widget.render({
+      results,
+      state: helper.state,
+      helper,
+      createURL: () => '#',
+    });
+
+    const rendering = renderFn.mock.calls[1][0];
+
+    expect(rendering.indices[0].hits).toEqual(
+      expect.arrayContaining(escapedHits)
```

Comment:
`expect.arrayContaining` is necessary because the structure is not exactly the same. I think the reason is because we internally set the `__escape` object to `true` on the `hits` array after we escape the hits.

I can add a comment about it.